tools/codegen/codespec/attribute.go

```diff
@@ -46,7 +46,10 @@ func (s *APISpecSchema) buildResourceAttr(name string, computability ComputedOpt
 	case OASTypeArray:
 		return s.buildArrayAttr(name, computability)
 	case OASTypeObject:
-		return nil, nil // TODO: add support for SingleNestedObject and MapAttribute
+		if s.Schema.AdditionalProperties != nil && s.Schema.AdditionalProperties.IsA() {
```

Comment:
q: what does `IsA()` imply?

Reply:
libopenapi stores additionalProperties as Dynamic values (`AdditionalProperties *DynamicValue[*SchemaProxy, bool] `) defined here https://github.com/pb33f/libopenapi/blob/eff4ce83e7814228bc50e34e5e60c4312a041e98/datamodel/high/base/dynamic_value.go#L20 "The left value (A) is a 3.0 schema property value"